src/components/calendar/Calendar.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import { useState } from "react";
+import CalendarEvent from "@/components/calendar/CalendarEvent";
+import { getEventData } from "@/hooks/events";
+
+export interface DataProps {
+  date_time: string;
+  date: string;
+  title: string;
+  date_utc: string;
+  url: string;
+  contact_info: string;
+  description: string;
+  description_long: string;
+}
+
+const Calendar = () => {
+  const [view, setView] = useState("upcoming");
+  const data = getEventData();
+
+  if (!data) return <p>No event data</p>;
+
+  return (
+    <section className="container px-2 my-6">
+      <div className="hidden min-h-8 relative lg:block">
+        <div className="toggle-btn">
+          <p
+            id="week"
+            className={view === "weekly" ? "selected" : ""}
+            onClick={() => setView("weekly")}
+          >
+            Weekly
+          </p>
+          <p
+            id="month"
+            className={view === "monthly" ? "selected" : ""}
+            onClick={() => setView("monthly")}
+          >
+            Monthly
+          </p>
+          <p
+            id="upcoming"
+            className={view === "upcoming" ? "selected" : ""}
+            onClick={() => setView("upcoming")}
+          >
+            Upcoming
+          </p>
+        </div>
+      </div>
+
+      <div className="flex flex-wrap justify-between gap-4">
+        {data &&
+          data.map((e: DataProps, i) => {
+            return (
+              <div key={i}>
```

Comment:
You're not technically supposed to use the index of a thing as it's key. Could you do something like this instead?

```suggestion
              <div key={`${e.date_time}-${e.contact_info}-${e.title}`}>
```

Reply:
How come? Is it because if you have more than one map the keys can be the same within the view?

Reply:
React uses the key of a fragment to cache the rendered output for the result of the map, and I'm pretty sure indexes aren't guaranteed to be the same (if you sort an array, the keys would no longer match up). Like, it works, but there's a performance hit. 

Here's a SO post on this if you're curious to read more! https://stackoverflow.com/a/59518082